packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -132,10 +132,10 @@ function ti(value: any): number {
 // [s]lot element node
 function s(
     slotName: string,
-    data: VElementData,
-    children: VNodes,
+    data: Readonly<VElementData>,
+    children: Readonly<VNodes>,
     slotset: SlotSet | undefined
-): VElement | VNodes {
+): VElement | Readonly<VNodes> {
```

Comment:
Adding `Readonly<VNodes>` all around in the codebase is not ideal. I think the best approach here is to change the `VNodes` type definition in `vnode.ts` to:

```ts
export type VNodes = Readonly<Array<VNode | null>>;
```

Reply:
Done. Great suggestion.